src/reportengine/app.py

```diff
@@ -350,6 +350,8 @@ def run(self):
                 traceback_if_debug(e)
             sys.exit(1)
 
+        c.dump_lockfile()
```

Comment:
Should we do this in Environment.init_output(). Does it matter?

Reply:
I think if we tried to do it in init_output() then it would be too early because the config hasn't been parsed yet, with some minor testing I put it as early as the lock file appeared to have been updated